addons/portal_rating/static/src/js/portal_rating_composer.js

```diff
@@ -97,13 +95,22 @@ publicWidget.registry.RatingPopupComposer = publicWidget.Widget.extend({
 
     /**
      * @private
-     * @param {OdooEvent} ev
+     * @param {OdooEvent} event
      */
-    _onReloadRatingPopupComposer: function (ev) {
-        this._reloadRatingPopupComposer(ev.data);
+    _onReloadRatingPopupComposer: function (event) {
+        const data = event.data;
+
+        // Refresh the internal state of the widget
+        this.rating_avg = data.rating_avg;
+        this.rating_count = data.rating_count;
+        this.rating_value = data.rating_value;
+
+        this.options = _.extend(this.options, data);
```

Comment:
Shouldn't we pop rating_* keys from data so that we are sure that either we have rating values, either options, but not both ? That way value storage could be cleaner. What do you think ?

Reply:
yes good idea :)